### ISSUE TITLE
Using center_x with extended size does not work

### ISSUE THREAD

**Issue by kjf** [colinta/teacup#86]
If I set a size in one style and then extend it in another then the center_x frame calculation does not work.

For example:

``` ruby
  style :button,
    size: [90, 90],
    backgroundColor: UIColor.redColor

  style :my_button, extends: :button,
    top: 0,
    center_x: 80, # <-- This results in a frame of [[80, 0], [90, 90]] instead of [[35, 0], [90, 90]]
    title: 'My Button'
```

**Comment by colinta** [comment]
I'm gonna try and take a look at this soon. Thanks!

**Comment by colinta** [comment]
I've got an idea on how to fix this, by introducing a kind of "dependencies" feature, which would have `:center_x` depend on `:size` (e.g., `:size` gets applied first, no matter where it is defined)

**Comment by colinta** [comment]
K, finally taking a stab at this now.  In the future, adding a test case to `spec/ios/` would be a HUGE help.

**Comment by colinta** [comment]
Resolved in 2.1.4.  Thanks for finding this one!
